freva-rest/src/freva_rest/databrowser_api/core.py

```diff
@@ -75,6 +78,72 @@ async def wrapper(*args: Any, **kwargs: Any) -> asyncio.Task[Any]:
     return wrapper
 
 
+
+def swap_my_back(desired_outputs: List[str]):
+    """Decorator that tries secondary backend first, with Solr as fallback.
+    This decorator has been created to handle the different RDBMS or SE backends
+    without cluttering the main codebase with different backends specific logic
+    and functions.
+    ACHTUNG: The side effect of this decorator is that when we change the name of
+    functions in the main codebase, we have to update the decorator as well manually.
+
+    Parameters
+    ----------
+    desired_outputs : List[str]
+        List of required outputs from the backend query
+
+    Returns
+    -------
+    Callable
+        Decorated function that handles secondary backend operations
+    """
+    def decorator(func: Callable) -> Callable:
+        @wraps(func)
+        async def wrapper(self, *args, **kwargs):
+            if self.backend_type in ("RDBMS", "SE"):
+                start_method_key = 'offset' if self.backend_type == 'RDBMS' else 'from'
```

Comment:
I really don't like this sort of comprehensions. Please write them in multiple lines.